code/game/machinery/teleporter.dm

```diff
@@ -406,37 +378,38 @@
 	tele_delay = max(A, 0)
 	update_icon()
 
+//Prevents AI cores from using the teleporter, prints out failure messages for clarity
+/obj/machinery/teleport/proc/blockAI(M as mob|obj)
```

Comment:
`M as mob|obj` is ugly. I would just do `atom/A` since you're checking for specific types below anyway